cdap-proto/src/main/java/io/cdap/cdap/proto/ApplicationUpdateDetails.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto;
+
+import io.cdap.cdap.api.common.HttpErrorStatusProvider;
+import javax.annotation.Nullable;
+
+/**
+ * Represents an application update result of an {@link ApplicationDetail}.
+ */
+public class ApplicationUpdateDetails {
+
+  private final int statusCode;
+  private final String error;
+  private final String updateDetails;
```

Comment:
do we need both an error field and this field? Seems like we could just have a 'message' field since the status code indicates where it succeeded or failed

Reply:
Intially I had just one field "message" which included but once I was testing it out, I thought of  adding a human readable status code too which made it better IMO so I kept it. Statuscode itself is sufficient but was not very non-eng friendly is my thought. Let me know wdyt.